app/Services/Registration/Factories/ClientUserFactory.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Services\Registration\Factories;
+
+use App\DTO\CreateClientUserResource;
+use App\Enums\UserRolesEnum;
+use App\Models\ClientUser;
+use App\Repositories\ClientUserRepository;
+use App\Services\Registration\Interfaces\Factories\ClientUserFactoryInterface;
+
+final class ClientUserFactory implements ClientUserFactoryInterface
+{
+    public function __construct(private ClientUserRepository $clientUserRepository)
```

Comment:
If we are doing a dependency injection we should instantiate under construct, for example. 

```suggestion
        private UserRepositoryInterface $userRepository;

    public function __construct(UserRepositoryInterface $userRepository) {
        $this->userRepository = $userRepository;
    }
```